src/config.ts

```diff
@@ -26,7 +28,29 @@ export async function getAwsConfigFromOptionsOrFile(options: {
   region: string;
 }): Promise<AWSConfig> {
   const { profile, accessKey, secretKey, sessionToken, region } = options;
+  if (profile){
+    const credentials = await fromSSO({
+      profile: profile,
+    });
 
+    const sso_credentials = new SSOClient({
+      credentials: credentials,
+      region: region,
+    });
+    const creds = await sso_credentials.config.credentials()
+    const accessKeyId = creds["accessKeyId"];
+    const secretAccessKey = creds["secretAccessKey"];
+    const sessionToken = creds["sessionToken"];
+    return {
+      credentials: {
+        accessKeyId: accessKeyId,
+        secretAccessKey: secretAccessKey,
+        sessionToken: sessionToken
+      },
+      region: region,
+    };
+  
+  }
```

Comment:
This will break the existing functionality. Can we put SSO retrieval as a last step when none of the other options end up returning the credentials.

Reply:
Also consider moving this to a separate function.

Reply:
separate function is created loadSsoAwsCredentials
 SSO retrieval is placed is last